### PR TITLE
Refactored Root Structure + Aliases + Future Prep

### PR DESCRIPTION
**Pull request by ap0nia**
## New Plan
- Currently contains the refactored root structure and some prep-work required for my future progressions.
- Will be merging this into main soon
- From that point onward, I'll create a feature branch and point it at main per usual, and because I'll be moving very quickly, I'll just stack/nest feature branches onto one another so there's a smaller diff between each branch. As each branch gets approved, they'll collapse into main as I crank out the overhaul 😎 

## Summary
Overhaul the current project to be implemented using modern React-TypeScript idiomatic practices and libraries.

### ~~Implementation Plan~~
~~- Target implementation is the [aponia](https://github.com/icssc/AntAlmanac/tree/aponia) branch~~
~~- Make branches off this overhaul branch and make pull requests to this one until it's ready to be shipped, similar to @EricPedley TypeScript conversion #411~~

### ~~Child PRs~~
- [ ] ~~Refactor the root file structure of the app, i.e. `index.jsx`, `App.jsx`, providers, theme, etc.~~
- [ ] ~~Refactor all of the imports to use aliased paths~~
- [ ] ~~Add Zustand stores for state management~~

## ~~Test Plan~~
~~Verify successful compilation on preview deployments and click around testing website features~~

## ~~Closes~~ Will be closed over course of future PRs
- #267 
- #279
- #335 because I'll be using [Chart.js](https://www.chartjs.org/) and [react-chartjs-2](https://react-chartjs-2.js.org/) lol
- #391 because [@tanstack/react-query](https://tanstack.com/query/latest/docs/react/overview) will handle all server-side state management extremely optimally
- #401 in general, this overhaul relies heavily on React hooks, a modern/recommended practice
- #466 because `withStyles` will be removed all together, replaced with the [`sx` prop and inline styling](https://mui.com/material-ui/customization/how-to-customize/#1-one-off-customization)

### CODE REVIEW

**Review by ap0nia — COMMENTED** [review]
🙏

**Review by EricPedley — CHANGES_REQUESTED** [review]
Preview deployment looks fine but when I try to run it locally it doesn't render anything and I get these errors:
image
Any ideas of why this would break a development environment? I'm concerned that if I'm having these problems then merging this would brick the rest of the team's development environments as well.

**Comment by ap0nia** [comment]
Works on my machine :tm:

Damn that might actually be indicative of one of the shittiest bugs I've encountered before lol. The issue I had was that something got screwed up on the package manager/cache level and then it literally doesn't like certain file names/path. Just changing the filename to something else fixed the error. I wasn't able to get the file names to be accepted even after attempting to reset node_modules, global yarn cache, etc. so I have no clue where the origin of it even was. You might need to reset the entire node install? (hopefully you're using `nvm` or `fnm` if you try that lol) 

(After doing `rm -rf node_modules`, `yarn install`, etc.) Could you test if simply renaming the files has any effect? And if it does, it might be the aforementioned issue.

**Comment by EricPedley** [comment]
```
> rm -rf node_modules
> yarn install
> still doesn't work
> rename analytics.ts -> analytics-test.ts
> even more console errors
> rename analytics-test.ts -> analytics.ts
> works all of a sudden???
```
~~Ok seems like it was some sort of nightmare package manager / build system caching problem. If anyone gets stuck on this they can probably get around it by re-installing packages.~~
Sike! It still is crashing, this time when I search something:
image

**Comment by ap0nia** [comment]
Does that happen on a fresh clone as well?

**Comment by EricPedley** [comment]
I figured it out. My browser extension `uBlock Origin` is blocking `analytics.ts` from loading. For whatever reason it _doesn't_ do this on `main` but does on this branch, and doesn't do it on staging deployments because everything is bundled. Can you try reproducing this with an adblocker and making it fail gracefully with a message like "try disabling your adblocker" or something?

**Comment by ap0nia** [comment]
I can't fix it? https://github.com/react-ga/react-ga/issues/449

Nobody's responded to this post; it doesn't seem like there's a way to check if ReactGA4 actually works before invoking it. There doesn't seem like a reliable solution to fix this; I literally removed all references to `React-GA4` and it still would get blocked, so this is a browser level shenanigan

**Comment by EricPedley** [comment]
Ok fuck it we'll just put a warning in the README. On a related note, is port 3000 the default development server for Vite? Whenever I run it locally I get port 5173 and I'm wondering if it's a _me_ problem or if we should update the docs for that too.

**Comment by ap0nia** [comment]
Port 5173 is default for Vite, apparently it's leetspeak for Vite lol. You or I can push a config update if you prefer port 3000
https://vitejs.dev/config/server-options.html#server-port

**Comment by EricPedley** [comment]
Eh I'll just leave it as the default and update the docs.

**Comment by ap0nia** [comment]
Idk if uOrigin was initially triggered simply because that file attempted to load analytics once and uOrigin remembers it as sus forever or because it's literally just looking at the file's name.

image

**Review by EricPedley — APPROVED** [review]
@bevm0 do the docs changes look good?

File: README.md [hunk 1]
```diff
@@ -54,13 +54,16 @@ If you ever need help, feel free to ask around on our [Discord server](https://d
 
 2. Navigate to the root directory and install the dependencies  
    `cd AntAlmanac`  
-   `npm install`
+   `npm i -g yarn` (if this fails, try running it with admin privileges)  
+   `yarn install`
 
 3. Start the development server  
    `npm start`
```
Comment (ap0nia) [inline README.md]: maybe `yarn start` just to be consistent? They all do the same thing, but might as well commit to using yarn for everything if we bring it up at the beginning

Comment (EricPedley) [reply README.md]: Agreed.

Comment (ap0nia) [reply README.md]: Also recommending admin privileges really rubs me the wrong way; running as admin is usually an escape hatch if you didn't set up your dev environment properly. It's the easiest work-around to get started, but you really should be using [fnm](https://github.com/Schniz/fnm) or [nvm](https://github.com/nvm-sh/nvm) to manage your node installation for the current user. But maybe it doesn't seem as beginner friendly to bring that up?

**Pull request merged**